reconcile/terraform_tgw_attachments.py

```diff
@@ -255,38 +267,46 @@ def run(
     print_to_file: Optional[str] = None,
     enable_deletion: bool = False,
     thread_pool_size: int = 10,
+    account_name: Optional[str] = None,
     defer: Optional[Callable] = None,
 ) -> None:
     settings = queries.get_secret_reader_settings()
     clusters = queries.get_clusters_with_peering_settings()
-    tgw_clusters = _filter_tgw_clusters(clusters)
+    tgw_clusters = _filter_tgw_clusters(clusters, account_name)
     ocm_map = _build_ocm_map(tgw_clusters, settings)
-    accounts = queries.get_aws_accounts(terraform_state=True, ecrs=False)
+    accounts = queries.get_aws_accounts(
+        terraform_state=True,
+        ecrs=False,
+        name=account_name,
+    )
```

Comment:
Since you are modifying this integration, isn't it a good time to move to typed queries instead of continue using the queries module?

Reply:
Yes, I'll work on that next, if the changes are too big, I'll open a new PR.

Reply:
@lechuk47 I just opened a [pr](https://github.com/app-sre/qontract-reconcile/pull/3467) to add one typed query for `queries.get_clusters_with_peering_settings`, there will be a lot of conflicts when replacing it with this pr, so I plan to merge both then do the switch. After that I can do the other 2 (`queries.get_secret_reader_settings` and `queries.get_aws_accounts`), what do you think?

Reply:
I can change it to DI way, just need some additional work to let mypy happy

Reply:
DI refactor blocked, see this [comment](https://github.com/app-sre/qontract-reconcile/pull/3467#discussion_r1173736976), so the plan is to merge these 2 PRs, do refactors one by one.

Reply:
typed query in https://github.com/app-sre/qontract-reconcile/pull/3470